waltid-wallet-api/src/main/kotlin/id/walt/webwallet/web/controllers/AuthController.kt

```diff
@@ -62,388 +61,455 @@ data class LoginTokenSession(val token: String) : Principal
-                            put("email", JsonPrimitive("user@email.com"))
-                            put("password", JsonPrimitive("password"))
-                            put("type", JsonPrimitive("email"))
-                        }.toString())
-                        example("Wallet address + ecosystem", buildJsonObject {
-                            put("address", JsonPrimitive("0xABC"))
-                            put("ecosystem", JsonPrimitive("ecosystem"))
-                            put("type", JsonPrimitive("address"))
-                        }.toString())
-                    }
-                }
-                response {
-                    HttpStatusCode.Created to { description = "Registration succeeded " }
-                    HttpStatusCode.BadRequest to { description = "Registration failed" }
-                }
-            }) {
-                val req = LoginRequestJson.decodeFromString<AccountRequest>(call.receive())
-                AccountsService.register("", req).onSuccess {
-                    call.response.status(HttpStatusCode.Created)
-                    call.respond("Registration succeeded ")
-                }.onFailure { call.respond(HttpStatusCode.BadRequest, it.localizedMessage) }
+            response {
+              HttpStatusCode.OK to { description = "Login successful" }
+              HttpStatusCode.Unauthorized to { description = "Login failed" }
+              HttpStatusCode.BadRequest to { description = "Login failed" }
             }
-
-            authenticate("auth-session", "auth-bearer", "auth-bearer-alternative") {
-                get("user-info", { summary = "Return user ID if logged in" }) {
-                    call.respond(getUserId().name)
+          }) {
+            val reqBody = LoginRequestJson.decodeFromString<AccountRequest>(call.receive())
+            AccountsService.authenticate("", reqBody)
+                .onSuccess { // FIXME -> TENANT HERE
+                  securityUserTokenMapping[it.token] = it.id
+                  call.sessions.set(LoginTokenSession(it.token))
+                  call.response.status(HttpStatusCode.OK)
+                  call.respond(
+                      mapOf(
+                          "token" to it.token,
+                          "id" to it.id.toString(),
+                          "username" to
+                              it.username // TODO: change id to wallet-id (also in the frontend)
+                          ))
                 }
-                get("session", { summary = "Return session ID if logged in" }) {
-                    val token = getUsersSessionToken() ?: throw UnauthorizedException("Invalid session")
-
-                    if (securityUserTokenMapping.contains(token)) call.respond(mapOf("token" to mapOf("accessToken" to token)))
-                    else throw UnauthorizedException("Invalid (outdated?) session!")
+                .onFailure {
+                  it.printStackTrace()
```

Comment:
I'd avoid using print-stack-trace
maybe `logger.error { it.stackTraceToString() }`?